### PR TITLE
Use sphinxcontrib-bibtex for references

### LINKED ISSUE #1726: Use sphinxcontrib-bibtex for all references in the docstrings
https://sphinxcontrib-bibtex.readthedocs.io/en/latest/index.html

### PR DESCRIPTION
**Pull request by martinkim0**
Closes #1726.

### CODE REVIEW

**Review by adamgayoso — COMMENTED** [review]
Should just need to add the bib file into conf.py?

File: pyproject.toml [hunk 1]
```diff
@@ -95,6 +96,7 @@ docs = [
   "sphinx-design",
   "sphinxext-opengraph",
   "sphinx-hoverxref",
+  "sphinxcontrib-bibtext",
```
Comment (adamgayoso) [inline pyproject.toml]:
```suggestion
  "sphinxcontrib-bibtex",
```

File: scvi/__init__.py [hunk 2]
```diff
@@ -3,12 +3,11 @@
 # Set default logging handler to avoid logging with logging.lastResort logger.
 import logging
 
+# this import needs to come after prior imports to prevent circular import
+from . import data, external, model, utils
```
Comment (adamgayoso) [inline scvi/__init__.py]: this needs to be kept after. Perhaps a no QA comment if it's an issue?

Comment (martinkim0) [reply scvi/__init__.py]: Yeah flake8 autofixed it for some reason...not sure why but when changing it back without noqa it seems to pass

Comment (adamgayoso) [reply scvi/__init__.py]: I would still add a noqa so it doesn't do it again

**Comment by adamgayoso** [comment]
actually can we tell flake to ignore that init file?

**Comment by martinkim0** [comment]
> actually can we tell flake to ignore that init file?

Wait I think flake8 already ignores `__init__` files:
```
per-file-ignores =
    tests/*: D
    */__init__.py: F401
```

**Review by adamgayoso — APPROVED** [review]

**Comment by adamgayoso** [comment]
Actually @martinkim0 there are a few issues with the bib file:

https://scvi--1731.org.readthedocs.build/en/1731/references.html

Can you please add the missing info? Then feel free to merge once it's fixed (also I added two commits, so git pull --rebase)

File: docs/references.bib [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+  number = {2},
+  pages = {163--166},
+  journal = {Nature Biotechnology}
+}
+
+@article{Lotfollahi21,
+  title = {Mapping single-cell data to reference atlases by transfer learning},
+  author = {Mohammad Lotfollahi and Mohsen Naghipourfar and Malte D. Luecken and Matin Khajavi and Maren B\"{u}ttner and Marco Wagenstetter and {\v{Z}}iga Avsec and Adam Gayoso and Nir Yosef and Marta Interlandi and Sergei Rybakov and Alexander V. Misharin and Fabian J. Theis},
+  doi = {10.1038/s41587-021-01001-7},
+  url = {https://doi.org/10.1038/s41587-021-01001-7},
```
Comment (adamgayoso) [inline docs/references.bib:33]: also the doi should be sufficient, we don't need doi and url

**Comment by martinkim0** [comment]
@adamgayoso Added the user guide diagram to the README, lmk what you think

**Pull request merged**